### PR TITLE
allow full mic device name , as prefix is changing from plughw to hw,…

### PR DESCRIPTION
**Pull request by sdetweil**
… don't break old apps

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,13 +29,17 @@ function PocketSphinxContinuous(config) {
 
 PocketSphinxContinuous.prototype.start = function() {
   if (!this.isListening()) {
+    let mic=this.microphone;
```
Comment (fewieden) [inline index.js]:
```suggestion
    let mic = this.microphone;
```

File: index.js [hunk 2]
```diff
@@ -29,13 +29,17 @@ function PocketSphinxContinuous(config) {
...
+    // if name starts with number
+    if (this.microphone.match(/^\d/))
```
Comment (fewieden) [inline index.js]: Can you please add brackets to the if statement, I'm not a fan of this syntax.

File: index.js [hunk 3]
```diff
@@ -29,13 +29,17 @@ function PocketSphinxContinuous(config) {
...
+      mic = 'plughw:'+this.microphone;
```
Comment (fewieden) [inline index.js]:
```suggestion
      mic = `plughw:${this.microphone}`;
```

File: index.js [hunk 4]
```diff
@@ -29,13 +29,17 @@ function PocketSphinxContinuous(config) {
...
+      mic = 'plughw:'+this.microphone;      
     this._psc = spawn('pocketsphinx_continuous', [
       '-adcdev',
-      `plughw:${this.microphone}`,
+      mic,
       '-inmic',
       'yes',
       '-lm',
-      `modules/MMM-voice/${this.setId}.lm`,
+      'modules/MMM-voice/${this.setId}.lm',
```
Comment (fewieden) [inline index.js]: Please revert this change, this is a bug.

**Review by fewieden — CHANGES_REQUESTED** [review]
Hi @sdetweil thanks for your addition to this project. Please apply the requested changes so i can merge your pull request.

Also there seems to be a merge conflict, please merge the current master branch into yours.

Thank you

File: index.js (hunk 3 above)
Comment (sdetweil) [reply index.js]: i don't see that in my branch
i see
```
     `modules/${this.setId}/${this.setId}.lm`,
```

File: index.js [hunk 5]
```diff
@@ -29,15 +29,19 @@ function PocketSphinxContinuous(config) {
 
 PocketSphinxContinuous.prototype.start = function() {
   if (!this.isListening()) {
+    let mic = this.microphone;
+    // if name starts with number
+    if (this.microphone.match(/^\d/))
+      mic = `plughw:+${this.microphone}`;
```
Comment (fewieden) [inline index.js]:
```suggestion
      mic = `plughw:${this.microphone}`; 
```

File: index.js (hunk 2 above)
Comment (fewieden) [reply index.js]: In the default config in the repo the microphone is a number, here you assume it is a string. But then the code will fail due to `TypeError: microphone.match is not a function`. See https://jsfiddle.net/38nwxmqs/

Comment (fewieden) [reply index.js]: I fixed it on your branch

File: index.js (hunk 5 above)
Comment (fewieden) [reply index.js]: Fixed it on your branch

**Review by fewieden — APPROVED** [review]

**Pull request merged**